tests/components/spotify/test_media_browser.py

```diff
@@ -1,58 +1,97 @@
 """Test the media browser interface."""
 
+from unittest.mock import MagicMock
+
 import pytest
 from syrupy import SnapshotAssertion
 
 from homeassistant.components.spotify import DOMAIN
 from homeassistant.components.spotify.browse_media import async_browse_media
+from homeassistant.const import CONF_ID
 from homeassistant.core import HomeAssistant
-from homeassistant.setup import async_setup_component
-
-from tests.common import MockConfigEntry
 
+from . import setup_integration
+from .conftest import SCOPES
 
-async def setup_integration(hass: HomeAssistant, config_entry: MockConfigEntry) -> None:
-    """Fixture for setting up the component."""
-    config_entry.add_to_hass(hass)
-
-    await hass.config_entries.async_setup(config_entry.entry_id)
-    await hass.async_block_till_done(wait_background_tasks=True)
-    await async_setup_component(hass, DOMAIN, {})
-    await hass.async_block_till_done(wait_background_tasks=True)
+from tests.common import MockConfigEntry
 
 
+@pytest.mark.usefixtures("setup_credentials")
 async def test_browse_media_root(
     hass: HomeAssistant,
+    mock_spotify: MagicMock,
+    mock_config_entry: MockConfigEntry,
     snapshot: SnapshotAssertion,
-    spotify_setup,
+    expires_at: int,
 ) -> None:
     """Test browsing the root."""
+    await setup_integration(hass, mock_config_entry)
+    # We add a second config entry to test that lowercase entry_ids also work
```

Comment:
Why do we test this? The entry_id is a random string assigned by Home Assistant when the config entry is created, why does this media browser test try both?

Reply:
https://github.com/home-assistant/core/pull/124368 because YARL takes everything lowercase and that breaks the media browser

Reply:
OK, I see